### PR TITLE
feat: page on pytest

### LINKED ISSUE #80: [page request] PyTest
It would be nice to have a pytest page, discussing how to do (proper) pytest, with parametrized fixtures and such.

### PR DESCRIPTION
**Pull request by henryiii**
Closes #80.

### CODE REVIEW

File: pages/developers/pytest.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+Tests are crucial to writing reliable software. A good test suite allows you to:
+
+* Immediately know if a new platform or software version works,
+* Refactor and cleanup your code with confidence,
+* Evaluate the effect of additions and changes.
+
+Python used to have three major choices for tests; but now [PyTest][] is used almost exclusively. Testing is never an install requirement, so there's no harm in using PyTest. The goals of writing good tests are:
+
+* Simplicity: the easer / nicer you tests are to write, the more you will write.
```
Comment (eduardo-rodrigues) [inline pages/developers/pytest.md:19]:
```suggestion
* Simplicity: the easer / nicer your tests are to write, the more you will write.
```

File: pages/developers/pytest.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+```toml
+[tool.pytest.ini_options]
+minversion = "6.0"
+addopts = "-ra"
+testpaths = [
+    "tests",
+]
+```
+
+The `minversion` will print a nicer error if your PyTest is too old (though, ironically, it won't read this is the version is too old, so setting "6" or less in `pyproject.toml` is rather pointless). The addopts setting will add whatever you put there to the command line when you run; `-ra` will print a summary "r"eport of "a"ll results, which gives you a quick way to review what tests failed and were skipped, and why. Finally, `testpaths` will limit pytest to just looking in the folders given - useful if it tries to pick up things that are not tests from other directories. [See the docs](https://docs.pytest.org/en/stable/customize.html) for more options.
```
Comment (eduardo-rodrigues) [inline pages/developers/pytest.md:58]:
```suggestion
The `minversion` will print a nicer error if your PyTest is too old (though, ironically, it won't read this is the version is too old, so setting "6" or less in `pyproject.toml` is rather pointless). The `addopts` setting will add whatever you put there to the command line when you run; `-ra` will print a summary "r"eport of "a"ll results, which gives you a quick way to review what tests failed and were skipped, and why. Finally, `testpaths` will limit PyTest to just looking in the folders given - useful if it tries to pick up things that are not tests from other directories. [See the docs](https://docs.pytest.org/en/stable/customize.html) for more options.
```

File: pages/developers/pytest.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+PyTest also checks the current and parent directories for a `conftest.py` file. If it finds them, they will get run top-to-bottom. These files let you add fixtures and other python configuration per-directory.
```
Comment (eduardo-rodrigues) [inline pages/developers/pytest.md:60]: "and other python configuration per-directory." does not read well and is confusing.

File: pages/developers/pytest.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+In general, do not place a `__init__.py` file in your tests; there's not often a reason to make the test directory importable, and it can confuse package discovery algorithms.
+
+### Running PyTest
+
+You can run pytest directly with `pytest` or `python -m pytest`. You can optionally give a directory or file to run on. You can also select just some subset of tests with `-k <expression>`, or an exact test with `filename.py::test_name`.
```
Comment (eduardo-rodrigues) [inline pages/developers/pytest.md:66]:
```suggestion
You can run PyTest directly with `pytest` or `python -m pytest`. You can optionally give a directory or file to run on. You can also select just some subset of tests with `-k <expression>`, or an exact test with `filename.py::test_name`.
```

**Review by eduardo-rodrigues — APPROVED** [review]
Cool!

File: pages/developers/pytest.md (hunk 3 above)
Comment (henryiii) [reply pages/developers/pytest.md]: Hopefully it’s better now, let me know if it is not!

Comment (eduardo-rodrigues) [reply pages/developers/pytest.md]: Yep.

**Pull request merged**